test/e2e/serial/tests/resource_accounting.go

```diff
@@ -159,9 +159,18 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload resourc
 			failedPodIds := e2ewait.ForPaddingPodsRunning(fxt, paddingPods)
 			Expect(failedPodIds).To(BeEmpty(), "some padding pods have failed to run")
 
-			for _, zone := range nrtInfo.Zones {
+			var targetNrtBefore *nrtv1alpha1.NodeResourceTopology
+			var targetNrtListBefore nrtv1alpha1.NodeResourceTopologyList
+			for idx, zone := range nrtInfo.Zones {
+				if idx == len(nrtInfo.Zones)-1 {
```

Comment:
this works. A better approach:

1. select the target node randomly (see note about reproducibility, however) earlier, around line 161
2. save the NRT counters state in this loop when `idx` matches the target node
3. delete the selected node using the pre-prepared `idx` down below around line 254

Reply:
IIUC and the target node you meant here is the target numa zone (because nrtInfo is the nrt of the target node), this scenario does not fit, because we don't want the selection of the numa to be random especially here, instead, we want it to be the last numa zone to hold the last placeholder pod in order to compare the NRTs before and after deleting that pod. saving NRTs upon scheduling earlier placeholder pods (hence numas) would lead to having an old version of the NRT, that when comparing it with the targetNrtAfter the result won't be accurate.

what can be done additionally is to compare the matching zone of which the NRT was updated (instead of expecting the match to be notEmpty), so I'll support this.